components/FormInput/FormInput.js

```diff
@@ -12,13 +13,14 @@ export default class FormInput extends Component {
     this.state = {
       active: !!props.value,
       focus: false,
+      id: uniqueId('forminput'),
     };
   }
 
   componentWillReceiveProps(nextProps) {
     if (!this.state.focus) {
       this.setState({
-        active: nextProps.value.length,
+        active: nextProps.value && nextProps.value.length,
```

Comment:
Smells like we're keeping state in sync with props. Should be calculated on the fly in `render`.